common/sys/platform.h

```diff
@@ -99,7 +99,7 @@
 #define dll_import 
 #endif
 
-#ifdef __WIN32__
+#if defined(__WIN32__) && !defined(__MINGW32__)
```

Comment:
The same might be relevant on line 94 for the definitions of `dll_export` and `dll_import`. But it's not part of the patch we're using in production for Godot (https://github.com/godotengine/godot/blob/master/thirdparty/embree/patches/godot-changes-misc.patch), so for this PR I chose to play it safe and only include what we tested.